packages/playback-core/src/index.ts

```diff
@@ -47,6 +57,12 @@ export const initialize = (
 ) => {
   // Automatically tear down previously initialized mux data & hls instance if it exists.
   teardown(mediaEl, core);
+  // NOTE: metadata should never be nullish/nil. Adding here for type safety due to current type defs.
+  const { metadata = {}, playbackId, src } = props;
+  const { view_session_id = generateUUID(), video_id = playbackId || src } = metadata;
```

Comment:
a session shouldn't change just because the user changes video id right?
calling `generateUUID()` at the top of the file seems more accurate to me

Reply:
The spec is a bit ambiguous, but it does seem like doing it per load is more helpful.
Spec says:

> A playback session typically ties together segments belonging to a single media asset.

Reply:
oh ok, my mind went right away to browser session. if it's clearly in the spec all good